From 1b11f166d25b65a9b2734f36effcacf43697b2f6 Mon Sep 17 00:00:00 2001
From: John Jacques <john.jacques@intel.com>
Date: Fri, 27 Oct 2017 12:06:14 -0500
Subject: [PATCH 5/7] pci-driver: HACK: merge for Altera...

For some reason, the previous changes required to sort out
the Altera FPGAs no longer works.  This is being debugged,
but for now, this commit gets to the Linux prompt.

Signed-off-by: John Jacques <john.jacques@intel.com>
---
 drivers/pci/pci-driver.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/pci/pci-driver.c b/drivers/pci/pci-driver.c
index 33442e8..95a8aa9 100644
--- a/drivers/pci/pci-driver.c
+++ b/drivers/pci/pci-driver.c
@@ -777,7 +777,7 @@ static int pci_pm_suspend_noirq(struct device *dev)
 
 	if (!pci_dev->state_saved) {
 		pci_save_state(pci_dev);
-		if (pci_power_manageable(pci_dev))
+		if (!pci_has_subordinate(pci_dev))
 			pci_prepare_to_sleep(pci_dev);
 	}
 
@@ -1144,6 +1144,7 @@ static int pci_pm_runtime_suspend(struct device *dev)
 		return -ENOSYS;
 
 	pci_dev->state_saved = false;
+	pci_dev->no_d3cold = false;
 	error = pm->runtime_suspend(dev);
 	if (error) {
 		/*
@@ -1160,6 +1161,8 @@ static int pci_pm_runtime_suspend(struct device *dev)
 
 		return error;
 	}
+	if (!pci_dev->d3cold_allowed)
+		pci_dev->no_d3cold = true;
 
 	pci_fixup_device(pci_fixup_suspend, pci_dev);
 
@@ -1395,6 +1398,8 @@ static int pci_uevent(struct device *dev, struct kobj_uevent_env *env)
 {
 	struct pci_dev *pdev;
 
+	return -ENODEV;
+
 	if (!dev)
 		return -ENODEV;
 
-- 
2.7.4

